lib/utility/app_router.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+
+import '../presentation/home_screen/home_screen.dart';
+import 'app_router_routes.dart';
+
+
+// example to use the app router :  Navigator.of(context).pushNamed(AppRoutes.homeScreen);
+class AppRouter {
+
+
+  Route? onGenerateRoute(RouteSettings settings) {
+    return MaterialPageRoute(builder: (context) {
+
+      // add new case with the name of your route and the screen you will navigate to
+
+      return Builder(
+        builder: (BuildContext context) {
+          switch (settings.name) {
+
+            case AppRoutes.homeScreen:
+              return const HomeScreen(title: 'Help',);
+
+
+
+              // in case the named route provided in not available
+            default:
```

Comment:
Maybe default to home & show a snackbar "Destination not found"?

Reply:
I thinks this method is better for debugging when going full release we need to change to your method .